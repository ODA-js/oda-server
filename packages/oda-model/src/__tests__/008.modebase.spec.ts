import 'jest';
import {
  ModelBase,
  modelBaseDefaultInput,
  modelBaseDefaultMetaInfo,
} from '../modelbase';
import { merge } from 'lodash';

describe('ModeBase', () => {
  it('default', () => {
    const res = new ModelBase({ name: 'A' });
    expect(res.metadata).toMatchObject(modelBaseDefaultMetaInfo);
    expect(res.toObject()).toMatchSnapshot('toObject');

    expect(res.toObject()).toMatchObject(
      merge({ name: 'A' }, modelBaseDefaultInput),
    );
  });
  it('do not change input name, title and descriptions start letters, but only trim it', () => {
    const res = new ModelBase({
      name: ' A ',
      title: ' a ',
      description: ' b ',
    });
    expect(res.name).toBe('A');
    expect(res.title).toBe('a');
    expect(res.description).toBe('b');
  });
});
