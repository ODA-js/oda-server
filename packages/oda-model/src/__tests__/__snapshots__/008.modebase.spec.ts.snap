// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModeBase default: toObject 1`] = `
Object {
  "description": undefined,
  "metadata": Object {},
  "name": "A",
  "title": undefined,
}
`;
